### DISCUSSION TITLE
Not getting GPS Last Reading iOS

### DISCUSSION THREAD

**Discussion by Dipannita20** [shinyorg/shiny#904] [Q&A] [answered]
iOS 15.3
**Location permission Deny**
`IGpsReading gpsReading = await GpsManager.GetLastReading().ToTask();`

Above line become infinite wait, it's neither throwing exception nor executing next line, any idea?

Thanks in advance

**Comment by aritchie** [comment, marked as answer]
There often isn't a "last reading"   right away.  You should always use a timeout before your ToTask.

**Reply by Dipannita20** [reply]
Thanks
